### DISCUSSION TITLE
Maximale URL-Länge prüfen

### DISCUSSION THREAD

**Discussion by fanie-mak** [e-netze/webgis-community#164] [General]
Die Idee wäre, dass WebGIS bzw. DataLinq beim Aufruf von Links die Maximale URL Länge prüft (Wir dürfen maximal 2046 Zeichen verwenden) und dann eine entsprechende Meldung ausgibt. 
- WebGIS <--> DataLinq
- DataLinq <--> DataLinq

Idee --> api.config neuer Key, zur Definition der maximalen Zeichenanzahl
Eine alternative Lösung wären Post-Aufrufe.

**Comment by gis-eni** [comment]
Das ist leider schwer möglich.

WebGIS: es wird kein Unterschied zu DataLinq und anderen Urls gemacht. Man würde hier Request zu anderen Diensten bescheiden, die einen längeren QueryString zulassen.

DataLinq: Im DataLinq baust du in Regel selber die Link Tags in ``.razor`` File bzw. in den ``Report``. DataLinq kann auch hier nicht für dich entscheiden was du willst. Du kannst das natürlich im Code selber abfragen und einen Fehler ausgeben, wenn dein Zielserver nicht mehr schafft.

Woher kommt diese Beschränkung?

**Comment by fanie-mak** [comment]
Hier einmal der Code für das Auslesen der URL-Länge:

var myURL = "......"
myURL_Laenge = myURL.length;

    if(myURL_Laenge >= 2048){
        alert("Die ausgewählte URL ist zu lang. Bitte weniger Grundstücke wählen.");
    }
    else{
        //do something else     
    }
